tests/py/dynamo/conversion/test_dropout_aten.py

```diff
@@ -0,0 +1,56 @@
+import torch
+import torch.nn as nn
+from parameterized import parameterized
+from torch.testing._internal.common_utils import run_tests
+
+from .harness import DispatchTestCase
+
+
+class TestDropOutConverter(DispatchTestCase):
+    @parameterized.expand(
+        [
+            ((10,), 0, False),
+            ((1, 3), 0.3, False),
+            ((2, 2, 2), 0.5),
+            ((2, 2, 2, 2), 1),
+        ]
+    )
+    def test_native_dropout(self, input_shape, p, train=False):
+        class NativeDropout(nn.Module):
+            def forward(self, input):
+                return torch.ops.aten.native_dropout.default(input, p, train)
+
+        inputs = [torch.randn(input_shape)]
+        self.run_test(
+            NativeDropout(),
+            inputs,
+        )
+
+    @parameterized.expand(
+        [
+            (
+                torch.randn(
+                    10,
+                ),
+                0,
+                False,
+            ),
+            (torch.randn(1, 3), 0.3, False),
+            (torch.randn(2, 2, 2), 0.5),
+            (torch.randn(2, 2, 2, 2), 1),
+        ]
+    )
+    def test_native_dropout_pytorch(self, input, p, train=False):
+        class NativeDropout(nn.Module):
+            def forward(self):
+                return torch.ops.aten.native_dropout.default(input, p, train)
+
+        inputs = []
```

Comment:
Could you please explain the reason for dividing the test cases into `inputs = [torch.randn(input_shape)]` and `inputs = []`?

Reply:
If passing in `inputs = [torch.randn(input_shape)]`, the input will be an ITensor. If passing the input directly into the pytorch function, the input will be a normal tensor.

Reply:
LGTM. The code looks good to merge.

I have one more question about input types for my understanding. Usually, we handle inputs as `input: TRTTensor` and only consider ITensor input. However, sometimes we use `input_val: Union[TRTTensor, torch.Tensor, np.ndarray]` to handle both ITensor and normal tensor types. Even though the schema doesn't show a difference, how do we distinguish between them?

Reply:
LGTM. The code looks good to merge.

I have one more question about input types for my understanding. Usually, we handle inputs as input: TRTTensor and only consider ITensor input. However, sometimes we use input_val: Union[TRTTensor, torch.Tensor, np.ndarray] to handle both ITensor and normal tensor types. Even though the schema doesn't show a difference, how do we distinguish between them?

Reply:
Yeah very good questions.

> 1. Even though the schema doesn't show a difference, how do we distinguish between them?

From my understanding, `ITensor` is a concept in TRT so PyTorch schema isn't able to distinguish them. (TRT is just one backend among many backends)

> 2. Usually, we handle inputs as input: TRTTensor and only consider ITensor input. However, sometimes we use input_val: Union[TRTTensor, torch.Tensor, np.ndarray] or input: Sequence[Union[TRTTensor, torch.Tensor, np.ndarray]] to handle both ITensor and normal tensor types.

I think in the entry point the inputs are not promised to be `ITensor`. Instead, they could be `torch.Tensor` or `np.ndarray`. So there are two ways to handle it. 1) use `enforce_tensor_types` to make it as `ITensor` 2) manually cast it to `ITensor` inside of the function. I think in most cases both are fine, but if this function is called by another function that passes in non-`ITensor`, the first solution will error out.

Reply:
I understand better now. Thank you so much for the explanation!